website/source/api/auth/gcp/index.html.md

```diff
@@ -0,0 +1,362 @@
+    --header "X-Vault-Token: ..." \
+    --request POST \
+    --data @payload.json \
+    https://vault.rocks/v1/auth/gcp/role/dev-role
+```
+
+## Read Role
+
+Returns the previously registered role configuration.
+
+| Method   | Path                         | Produces               |
+| :------- | :--------------------------- | :--------------------- |
+| `GET`   | `/auth/gcp/role/:role`        | `200 application/json` |
+
+### Parameters
+
+- `role` `(string: <required>)` - Name of the role.
+
+### Sample Request
+
+```
+$ curl \
+    --header "X-Vault-Token: ..." \
+    https://vault.rocks/v1/auth/gcp/role/dev-role
+```
+
+### Sample Response
+
+```json
+{
+  "auth":null,
+  "warnings":null,
+  "data":{
+    "max_jwt_exp": 900,
+    "max_ttl": 0,
+    "ttl":0,
+    "period": 0,
+    "policies":[
+      "default",
+      "dev",
+      "prod"],
+    "project_id":"project-123456",
+    "role_type":"iam",
+    "service_accounts": [
+      "dev-1@project-123456.iam.gserviceaccount.com",
+      "dev-2@project-123456.iam.gserviceaccount.com",
+      "123456789",
+    ]
+  },
+  "wrap_info":null,
+  "lease_duration": 0,
+  "renewable": false,
+  "lease_id": ""
+}
+
+```
+
+## List Roles
+
+Lists all the roles that are registered with the backend.
+
+| Method   | Path                         | Produces               |
+| :------- | :--------------------------- | :--------------------- |
+| `LIST`   | `/auth/gcp/roles`            | `200 application/json` |
+
+### Sample Request
+
+```
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request LIST \
+    https://vault.rocks/v1/auth/gcp/roles
+```
+
+### Sample Response
+
+```json  
+{
+  "auth": null,
+  "warnings": null,
+  "data": {
+    "keys": [
+      "dev-role",
+      "prod-role"
+    ]
+  },
+  "lease_duration": 0,
+  "renewable": false,
+  "lease_id": ""
+}
+```
+
+## Delete Role
+
+Deletes the previously registered role.
+
+| Method   | Path                         | Produces               |
+| :------- | :--------------------------- | :--------------------- |
+| `DELETE` | `/auth/gcp/role/:role`       | `204 (empty body)`     |
+
+### Parameters
+
+- `role` `(string: <required>)` - Name of the role.
+
+### Sample Request
+
+```
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request DELETE \
+    https://vault.rocks/v1/auth/gcp/role/dev-role
+```
+
+## Login
+
+Fetch a token. This endpoint takes a signed JSON Web Token (JWT) and
+a role name for some entity. It verifies the JWT signature to authenticate that
+entity and then authorizes the entity for the given role.
+
+| Method   | Path                         | Produces               |
+| :------- | :--------------------------- | :--------------------- |
+| `POST`   | `/auth/gcp/login`            | `200 application/json` |
+
+### Sample Payload
+
+- `role` `(string: "")` - Name of the role against which the login is being
+  attempted.
+- `jwt` `(string: "")` - Signed JSON Web Token ([JWT](https://tools.ietf.org/html/rfc7519)).
```

Comment:
This is more of a personal preference but, maybe hyperlink "JSON Web Token" directly: `Signed [JSON Web Token](https://tools.ietf.org/html/rfc7519).`